lib/httpClient.js

```diff
@@ -42,13 +42,13 @@ function Client (opts) {
   // used for forcing reconnects
   this.reconnectRate = this.opts.reconnectRate
 
-  this.resData = new Array(this.opts.pipelining)
-  for (let i = 0; i < this.resData.length; i++) {
-    this.resData[i] = {
+  this.resData = []
+  for (let i = 0; i < this.opts.pipelining; i++) {
+    this.resData.push({
       bytes: 0,
       headers: {},
       startTime: [0, 0]
-    }
+    })
```

Comment:
This array is pre-allocated as a memory optimization. Maybe we should clean it up and shrinked when destroy() is called.

Reply:
That might be true for c/c++. Here is a [benchmark](http://jsperf.com/literal-vs-new-23). Literal [] is faster than new.

I'll let you take a stab at it. I got this error on node v10.14 and 10.16.

`new Array(10) => ['10'] instead of [](10)`

Reply:
That's not what I mean. The difference is in `.push()`, not on creation.

Reply:
Sorry for the confusion. I was looking at speed instead of memory.

Would this be better?
```
this.resData = new Array(this.opts.pipelining)
for (let i = 0; i < this.opts.pipelining; i += 1) {
  this.resData[i] = {
    bytes: 0,
    headers: {},
    startTime: [0, 0]
  }
}
```